### PR TITLE
Fix STP RefboxGameStatePlaySelectionTest

### LINKED ISSUE #1330: Renable/Fix STPRefboxGameStatePlaySelectionTest
### Description of the task
STPRefboxGameStatePlaySelectionTest are commented out because they don't pass (#674). There are also outstanding TODOs in ai/hl/stp/stp_refbox_game_state_play_selection_test.cpp. The goal here is to fix the test cases, so that we can reenable the tests. Talk to @jordanzliu if there are any questions

### Acceptance criteria

- [ ] Complete all outstanding TODOs in ai/hl/stp/stp_refbox_game_state_play_selection_test.cpp
- [ ] Fix the test cases
- [ ] Reenable the tests

### PR DESCRIPTION
**Pull request by kelvinkoon**
### Description

Re-enabled tests in `stp_refbox_game_state_play_selection_test.cpp` and ensure they pass.

### Testing Done

Tests in `stp_refbox_game_state_play_selection_test.cpp` pass.

### Resolved Issues

Resolves #1330 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by jonathanlew — COMMENTED** [review]
Will let @jordanzliu and @akhilveeraghanta review first

File: src/software/ai/hl/stp/stp_refbox_game_state_play_selection_test.cpp [hunk 1]
```diff
@@ -225,23 +222,55 @@ class STPRefboxGameStatePlaySelectionTest
...
-// TODO (Issue #1330): Reenable these tests
-// INSTANTIATE_TEST_CASE_P(AllRefboxGameStates, STPRefboxGameStatePlaySelectionTest,
-//                        ::testing::ValuesIn(all_refbox_game_states.begin(),
-//                                            all_refbox_game_states.end()));
+// NORMAL_START is omitted since there is no preceding PREPARE state
+INSTANTIATE_TEST_CASE_P(
+    AllRefboxGameStates, STPRefboxGameStatePlaySelectionTest,
+    ::testing::Values(
+        std::make_tuple(RefboxGameState::HALT,
+                        Ball(Point(0, 0), Vector(), Timestamp::fromSeconds(0))),
```
Comment (akhilveeraghanta) [inline src/software/ai/hl/stp/stp_refbox_game_state_play_selection_test.cpp]: what's the point of parameterizing the ball? 

I just ran these tests locally, and it seems like any Position + Initial velocity doesn't effect the `calculateNewPlay` function?

I can set all of the parameters to `Ball(Point(0, 0), Vector(), Timestamp::fromSeconds(0)))` and the test still passes

Comment (kelvinkoon) [reply src/software/ai/hl/stp/stp_refbox_game_state_play_selection_test.cpp]: There was a TODO as follows:
`// TODO (Issue #1330): replace the ball with real parameterized values`

Previously, the parameters were all set to the default `Ball`.

Comment (akhilveeraghanta) [reply src/software/ai/hl/stp/stp_refbox_game_state_play_selection_test.cpp]: Right, but just curious what these parameterized values are supposed to test? The test still passes if I just replace all of the positions with `Ball(Point(0, 0), Vector(), Timestamp::fromSeconds(0)))`

File: src/software/ai/hl/stp/stp_referee_command_play_selection_test.cpp [hunk 2]
```diff
@@ -224,23 +221,55 @@ class STPRefereeCommandPlaySelectionTest
...
+                        Ball(Point(4.3, 2.8), Vector(), Timestamp::fromSeconds(0))),
+        std::make_tuple(RefereeCommand::INDIRECT_FREE_US,
+                        Ball(Point(-2, 2.8), Vector(), Timestamp::fromSeconds(0))),
+        std::make_tuple(RefereeCommand::INDIRECT_FREE_THEM,
+                        Ball(Point(2, -2.8), Vector(), Timestamp::fromSeconds(0))),
+        std::make_tuple(RefereeCommand::TIMEOUT_US,
+                        Ball(Point(0, 0), Vector(), Timestamp::fromSeconds(0))),
+        std::make_tuple(RefereeCommand::TIMEOUT_THEM,
+                        Ball(Point(0, 0), Vector(), Timestamp::fromSeconds(0))),
+        std::make_tuple(RefereeCommand::GOAL_US,
```
Comment (jordanzliu) [inline src/software/ai/hl/stp/stp_referee_command_play_selection_test.cpp]: I don't think it makes sense for `GOAL_*` states to have a selected play

**Review by jordanzliu — CHANGES_REQUESTED** [review]

File: src/software/ai/hl/stp/stp_refbox_game_state_play_selection_test.cpp (hunk 1 above)
Comment (kelvinkoon) [reply src/software/ai/hl/stp/stp_refbox_game_state_play_selection_test.cpp]: @jordanzliu Can you elaborate on what that TODO was intended for?

File: src/software/ai/hl/stp/stp_referee_command_play_selection_test.cpp (hunk 2 above)
Comment (kelvinkoon) [reply src/software/ai/hl/stp/stp_referee_command_play_selection_test.cpp]: Removed `GOAL_US` and `GOAL_THEM`.

File: src/software/ai/hl/stp/stp_refbox_game_state_play_selection_test.cpp (hunk 1 above)
Comment (akhilveeraghanta) [reply src/software/ai/hl/stp/stp_refbox_game_state_play_selection_test.cpp]: Looks like plays based on RefboxGameState will be calculated regardless of the position of the ball. 

I.e if we receive `KICKOFF_US` we don't really care if the ball is in the centre, we still obey the rules.

So I think you can safely un-parameterize the ball and just leave it default for these tests.

Comment (kelvinkoon) [reply src/software/ai/hl/stp/stp_refbox_game_state_play_selection_test.cpp]: Ball has been un-parametrized.

**Review by jordanzliu — COMMENTED** [review]
💯

**Review by jordanzliu — APPROVED** [review]
💯

**Review by jonathanlew — APPROVED** [review]

**Pull request merged**